consensus/state.go

```diff
@@ -1438,7 +1456,14 @@ func (cs *ConsensusState) addProposalBlockPart(msg *BlockPartMessage, peerID p2p
 	}
 	if added && cs.ProposalBlockParts.IsComplete() {
 		// Added and completed!
-		_, err = cdc.UnmarshalBinaryReader(cs.ProposalBlockParts.GetReader(), &cs.ProposalBlock, int64(cs.state.ConsensusParams.BlockSize.MaxBytes))
+		maxDataSize := cs.state.ConsensusParams.BlockSize.MaxTxsBytes
+		// header + evidence + last commit
+		const maxOtherPartsSize = 5000000 // 5 MB
```

Comment:
this is ugly. and it should be lower

Reply:
We should also calculate the exact amino overhead. (Perhaps we make a method in amino to reflect through all the structs and tell you how big it would be if you were to marshall it)

Reply:
we should be able to be more precise, and we should do the opposite (ie. subtract from maxBlockSize to get maxDataSize)